src/tools/kdb/command.h

```diff
@@ -0,0 +1,177 @@
+				       "Give debug information or ask debug questions (in interactive mode).", KEY_META, "opt", "d",       \
+				       KEY_META, "opt/long", "debug", KEY_META, "opt/arg", "none", KEY_END));                              \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/verbose", KEY_META, "description", "Explain what is happening.", KEY_META, "opt",     \
+				       "v", KEY_META, "opt/long", "verbose", KEY_META, "opt/arg", "none", KEY_END));                       \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/version", KEY_META, "description", "Print version info.", KEY_META, "opt", "V",       \
+				       KEY_META, "opt/long", "version", KEY_META, "opt/arg", "none", KEY_END));                            \
+	ksAppendKey (baseSpec, keyNew (baseKeyName "/color", KEY_META, "description", "Print never/auto(default)/always colored output.",  \
+				       KEY_META, "opt", "C", KEY_META, "opt/arg/help", "WHEN", KEY_META, "opt/long", "color", KEY_META,    \
+				       "opt/arg", "required", KEY_END));                                                                   \
+	ksAppendKey (baseSpec,                                                                                                             \
+		     keyNew (baseKeyName "/nonewline", KEY_META, "description", "Suppress the newline at the end of the output.",          \
+			     KEY_META, "opt", "n", KEY_META, "opt/long", "no-newline", KEY_META, "opt/arg", "none", KEY_END));             \
+	ksAppendKey (spec, keyNew (COMMAND_SPEC_KEY (COMMAND_NAME) "/nullterm", KEY_META, "description", "Use binary 0 termination.",      \
+				   KEY_META, "opt", "0", KEY_META, "opt/long", "null", KEY_META, "opt/arg", "none", KEY_END));
+
+#define GET_OPT_KEY(options, key) ksLookupByName (options, key, 0)
+#define GET_OPT(options, key) keyString (GET_OPT_KEY (options, key))
+
+#define HAS_ERR(errorKey) keyGetMeta (errorKey, "error/reason") != NULL
+#define HAS_ERR_CODE(errorKey) keyGetMeta (errorKey, "error/number") != NULL
+#define GET_ERR(errorKey) keyString (keyGetMeta (errorKey, "error/reason"))
+#define GET_ERR_CODE(errorKey) keyString (keyGetMeta (errorKey, "error/number"))
+#define GET_ERR_DESC(errorKey) keyString (keyGetMeta (errorKey, "error/description"))
+#define COMMAND_BASE_KEY(name) CLI_BASE_KEY "/" name
+#define COMMAND_SPEC_KEY(name) "spec:" COMMAND_BASE_KEY (name)
+
+// only use in the context of a command/sub-command (options variable and GET_OPTION_KEY macro have to be in scope)
+#define GET_BASIC_OPTIONS                                                                                                                  \
+	bool debug = false;                                                                                                                \
+	Key * tmp = GET_OPTION_KEY (options, "debug");                                                                                     \
```

Comment:
Based on https://github.com/ElektraInitiative/libelektra/pull/4943/files#r1208288747, this line is wrong. But I'm not actually sure how fix it. You could do 

```c
Key * tmp = GET_OPTION_KEY (options, COMMAND_BASE_KEY (COMMAND_NAME) "debug");
```

and it should work, but it would make the macro even harder to use, because now the `COMMAND_NAME` macro needs to be defined. You could also add a `name` parameter to the macro and use

```c
Key * tmp = GET_OPTION_KEY (options, COMMAND_BASE_KEY (name) "debug");
```

which is a bit easier to use, but I actually think the proper solution is using:

```c
Key * tmp = GET_OPTION_KEY (options, CLI_BASE_KEY  "/debug");
```

**and** instead of using the `ADD_BASIC_OPTIONS` macro to add the `--debug`, `--verbose` etc. options to all commands, you add them to the base `KeySet * options` in `main()`. The difference would be:

```c
# works with old kdb, and with current PR, but not with change proposed above
kdb get --verbose

# works with old kdb, and with change proposed above, but not with current PR
kdb --verbose get
```

That's because when using sub-commands with `elektraGetOpts` the location of options like `--verbose` relative to command arguments matters (same as with e.g. `git`). Personally, I think the `kdb --verbose get` option actually makes more sense, since `--verbose` is an option that is not specific to `get`.

Reply:
I see why this is a little confusing, I'll document it better,
```c
#define GET_OPTION_KEY(options, name) GET_OPT_KEY (options, COMMAND_BASE_KEY (COMMAND_NAME) "/" name)
```
is defined in every command file, so `GET_OPTION_KEY` in `GET_BASIC_OPTIONS` is different in every command file. `command.h` doesn't define GET_OPTION_KEY.